HealthcareSystem/PatientWebApp/Controllers/SurveyController.cs

```diff
@@ -45,15 +45,19 @@ public ActionResult AddSurvey(SurveyDTO surveyDTO)
                 _surveyService.AddSurvey(SurveyMapper.SurveyDTOToSurvey(surveyDTO));
                 _examinationService.CompleteSurveyAboutExamination(surveyDTO.ExaminationId);
                 return Ok();
-            }
-            catch (ValidationException exception)
+            }         
+            catch (NotFoundException exception)
             {
-                return BadRequest(exception.Message);
+                return NotFound(exception.Message);
```

Comment:
I would probably still return BadRequest here, not NotFound. NotFound generally only makes sense to me when the user is explicitly trying to fetch a resource which doesn't exist, which isn't the case here.